### PR TITLE
Closes #175 - Switch to get_admiral_option() for subject_keys

### LINKED ISSUE #175: Global option for subject keys
See https://github.com/pharmaverse/admiral/issues/1338 - similar update will be needed for admiralonco functions using subject_keys argument. Someone from the core dev team will take this on across admiral and admiralonco.

### PR DESCRIPTION
**Pull request by zdz2101**
Please add "Closes #<issue number>" to the title of the pull request. Then the
issue is closed automatically once it is merged to `main`.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

The following line requests to update the man pages by the "Man Pages" workflow.

/roxygenize

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@zdz2101 please update the NEWS.md file. @bundfussr hope you're ok to review this PR?

File: R/admiralonco-package.R [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #' @importFrom admiral censor_source count_vals derive_param_first_event
 #'   derive_var_obs_number derive_vars_merged event_source filter_confirmation
 #'   filter_extreme filter_relative list_tte_source_objects max_cond min_cond
-#'   params restrict_derivation yn_to_numeric
+#'   params restrict_derivation yn_to_numeric get_admiral_option set_admiral_options
```
Comment (bundfussr) [inline R/admiralonco-package.R:6]: Could you add the new functions such that all the functions are in alphabetic order?

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by bundfussr** [comment]
@zdz2101 , could you fix the styler check? The R-CMD checks should pass when re-running as they were caused by an issue in admiral.

**Comment by zdz2101** [comment]
@bundfussr Made the appropriate styler edits but now running into NAMESPACE issues because of the removal of `set_dataset()` from `admiraldev` which also impacts `signal_crpr()`/`get_crpr_dataset()` under `R/check_crpr.R` This may also be related to perhaps implementing an `admiralonco_environment` object to set up the object `crpr`.

**Comment by bundfussr** [comment]
@zdz2101 , yes, only the failing "Man Pages" check is not related to https://github.com/pharmaverse/admiralonco/pull/192.

**Pull request merged**